sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -154,8 +155,19 @@ export function createKeyVaultChallengeCallbacks(
       verifyChallengeResource(scope, request);
     }
 
+    const { error, claims: base64EncodedClaims } = parsedChallenge;
```

Comment:
Looks like you won't get this far when given a CAE challenge because you threw on line 151? A CAE challenge doesn't necessarily include a resource/scope